### ISSUE TITLE
read.ctd.ssda() problem reading longitude

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#2227]
I upgraded my R to 4.4.1 (I didn't notice that it came out a while back) and see an error.  This does not get exercised by the remote tests, because this data file is not included with oce.  (We have quite a lot of tests that do not get run on those test machines, because they are slow or require large files ... both of which are problems for CRAN.)

I'll look at this today or tomorrow.  There is not a huge rush but I do want to submit to CRAN in 10 days, so for sure I want to fix it up in the next couple of days.

```R
── Failed tests ──────────────────────────────────────
Failure (test_ctd_ssda.R:19:13): read.ctd.ssda() works (discussions/1929)
ctd[["longitude"]] (`actual`) not equal to 15.9990216666667 (`expected`).

  `actual`: NA
`expected`: 16
```

**Comment by dankelley** [comment]
Hm.  It gets lat, but not lon.
```R
str(d@metadata)
...
 $ latitude         : num 55.3
 $ longitude        : num NA
```

**Comment by dankelley** [comment]
Hm, they call it "Long", not "Lon".
```
➜  local_data git:(develop) ✗ grep -i lat ctd_ssda.csv
002 GPS003 003  N  Lat_1    -  -5.24288000000E+0005  1.00000000000E+0000  0.00000000000E+0000  0.00000000000E+0000  0.00000000000E+0000  0.00000000000E+0000
002 GPS003 004  N  Lat_2    -   0.00000000000E+0000  1.00000000000E-0006  0.00000000000E+0000  0.00000000000E+0000  0.00000000000E+0000  0.00000000000E+0000
; Datasets    Druck    Temp.    Leitf    RawO2    Boden    SALIN    SIGMA    AO2_%    AO2mg    Licor     IntDT     IntDT     Lat         Long       BsFlo    AO2ml
➜  local_data git:(develop) ✗ grep -i lon ctd_ssda.csv
002 GPS003 005  N  Long1    -  -5.24288000000E+0005  1.00000000000E+0000  0.00000000000E+0000  0.00000000000E+0000  0.00000000000E+0000  0.00000000000E+0000
002 GPS003 006  N  Long2    -   0.00000000000E+0000  1.00000000000E-0006  0.00000000000E+0000  0.00000000000E+0000  0.00000000000E+0000  0.00000000000E+0000
; Datasets    Druck    Temp.    Leitf    RawO2    Boden    SALIN    SIGMA    AO2_%    AO2mg    Licor     IntDT     IntDT     Lat         Long       BsFlo    AO2ml
```

**Comment by dankelley** [comment]
Hm.  Maybe I'm mixed up here.  The code is checking for file format by checking the first line with
```R
    if (missing(format)) {
        format <- if (grepl("^\\[cast header\\]", lines[1])) {
            1L
        } else if (grepl(",", lines[1])) {
            2L
        } else {
            stop("cannot determine file 'format' by examining first line (shown below)\n", lines[1])
        }
        oceDebug(debug, "inferred format=", format, " from file's first line\n")
    }
```
but the first 3 lines are as follows, so the first line does not match either thing I'm demanding, and therefore should give an error.  But the test is not reporting that error.  (Also, I think I may need to tweak the encoding because notice there is a weird symbol in the first line, which I guess is an apostrophe.)
```
head -3 ../tests/testthat/local_data/ctd_ssda.csv
SSDA Sea & Sun Technology�s Standard Data Acquisition software package rev 1.83 17.08.2006
D:\SOFTWARE\SSDA_183\CONFIG\MOC002_GPS_04022021.SPJ
Montag, 19. April 2021 05:49:05
```

**Comment by dankelley** [comment]
Jeeze, I'm looking at the wrong file.  I'm going to quite tryting to read this small monitor and look at this issue tomorrow on a larger screen.  (There will be no power at home tomorrow anyway, due to some kind of transformer work.)

**Comment by dankelley** [comment]
Oh, the problem is that it is getting confounded by an "E" in the longitude field.  For the latitude field, it trims the "N".  Clearly I wrote that code in a rush, and without finishing things properly.

I will make it so that longitude can be any of the following
* a number
* a number with E at the end
* a number with W at the end
and similarly for latitude.  I guess I'll check for lower-case letters also.  I'll do it tomorrow morning.

**Comment by dankelley** [comment]
Fixed in "develop" branch, commit 6e1c5ebd517f9203b939b6b25747225c3c3ef802.
